webapp/frontend/lib/common/version_service.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+class Version {
+  final String version;
+
+  const Version({required this.version});
+
+  factory Version.fromJson(Map<String, dynamic> json) {
+    return switch (json) {
+      {
+        'version': String version,
+      } =>
+        Version(version: version),
+      _ => throw const FormatException('Failed to load version.'),
+    };
+  }
+}
+
+class VersionService {
+  const VersionService();
+
+  Future<String> fetchVersion() async {
+    // TODO(holzeis): this should come from the config
+    const port = "3001";
+    const host = "localhost";
```

Comment:
how do you see this evolve to support relative paths from where it's being hosted from?

Reply:
I fear we can't the URI path argument required by the http package does not support relative paths. At least I haven't found a way of doing that. I presume we will have to set that in a config. (see todo).